Makefile

```diff
@@ -51,6 +51,9 @@ validate:
 addons:
 	KUBECONFIG="$$(pwd)/phase1/$$(jq -r '.phase1.cloud_provider' .config.json)/.tmp/kubeconfig.json" ./phase3/do deploy
 
+deploy: | deploy-cluster validate addons
```

Comment:
Without this, `make` will try to execute the targets in parallel. There's another trick to disable all parallelism for the `Makefile` but apparently this enforces serial behavior for a single target.